### DISCUSSION TITLE
Remote Control (HmIP RC-8) button press event problem and solution

### DISCUSSION THREAD

**Discussion by dayt47** [SukramJ/aiohomematic#869] [General]
Hello,

I tried the following procedure to see the button presses of my 8 channel remote control (HmIP RC-8) in Home Assistant: [https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices](url)

It didn't work for me. But what helped is that I created a program in the Homematic / Raspberrymatic like the following picture:

<img width="1192" alt="fb8 channel home assistant solution" src="https://user-images.githubusercontent.com/32074282/213504557-ce408777-cb16-424d-a119-de1d97868395.png">

The program should stay deactivated and it still works in this way when you press a button, you will se the event in the logbook of Home Assistant. It also works with long press events although only short press is triggered in the picture.

@danielperna84 maybe you could add this solution to the page: [https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices](url)

P.s. When deleting the automation in Homematic / Raspberrymatic the button press events won't work in Home Assistant.

**Comment by Baxxy13** [comment]
Seit CCU3/RM Firmware 3.67.10 sind Punkt 8 & 9 der "Anleitung" nicht mehr korrekt.
Jeder Kanal (für den die Events aktiviert werden sollen) muss jetzt im Programm aufgenommen werden und dort bleiben.
(so wie es der Screenshot von @dayt47 zeigt)
Das Programm darf nicht gelöscht werden, kann aber auf inaktiv gestellt werden.  

Für RaspberryMatic braucht man aber kein "Dummy-Programm" mehr um die Events von IP-Tasterkanälen zu aktivieren. 
Das kann man direkt in den Kanaleigenschaften machen. 
![RM_Tasterevent_Kanal_OC8](https://user-images.githubusercontent.com/54898249/213524062-6b4ce6fb-7f03-49a6-8bda-dab8dab89820.JPG)

**Comment by SukramJ** [comment]
Vielen Dank für die Änderungsvorschläge.
Da es sich um ein Community-Projekt handelt kann jeder die Änderungen, in diesem Fall der README Datei, vornehmen und dann als  Pull Request einstellen. Andere können sich dann an dem PR beteiligen, indem direkt im PR Änderungswünsche angemerkt, bzw. Zustimmung oder Ablehnung ausgedrückt werden.
Letztendlich müssen @danielperna84 oder ich diese Änderungen dann nur durch einen Klick übernehmen.
